src/Symfony/Component/RateLimiter/Limit.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * @author Valentin Silvestre <vsilvestre.pro@gmail.com>
+ *
+ * @experimental in 5.2
+ */
+final class Limit
+{
+    private $availableTokens;
+    private $retryAfter;
+    private $accepted;
+    private $strategy;
+    private $metadata;
+
+    public function __construct(int $availableTokens, \DateTimeImmutable $retryAfter, bool $accepted, string $strategy, array $metadata = [])
+    {
+        $this->availableTokens = $availableTokens;
+        $this->retryAfter = $retryAfter;
+        $this->accepted = $accepted;
+        $this->strategy = $strategy;
```

Comment:
Instead of passing $strategy and $metadata, what about passing the LimiterInterface itself and use it?

```php
public function __construct(int $availableTokens, \DateTimeImmutable $retryAfter, bool $accepted, LimiterInterface $limiter)
{
    $this->availableTokens = $availableTokens;
    $this->retryAfter = $retryAfter;
    $this->accepted = $accepted;
    $this->limiter = $limiter;
}

public function getStrategy(): string
{
    return $this->limiter->getStrategyName();

    // or use the FQCN?
    return get_class($this->limiter);
}

public function getMetadata(): array
{
    return $this->limiter->getMetadata();
}
```

@wouterj, WDYT?

Reply:
The memory footprint would increase a lot imo, I'm waiting for wouterj opinion

Reply:
Btw, I messed up with my commits. I hardcoded the string in construct, I do not really love it but I think const are needed there

Reply:
Hmm, I don't see much advantage of doing so.

It would also bring a service (which has a storage injected, which has a storage driver - e.g. cache or a redis connection or whatever - injected) in a "simple" data object. I would be in favor of keeping the data object as simple as possible.

Reply:
Fair enough. My only thought was to reduce the number constructor arguments.

Reply:
I would suggest using the FQCN for the strategy name. Would make using a switch to determine the how to output headers easier:

```php
$limit = $limiter->consume();

switch ($limit->getStrategy()) {
    case TokenBucketLimiter::class:
        // ...
    case FixedWindowLimiter::class:
        // ...
}
```

WDYT?

Reply:
If using a FQCN, the only way to enforce this is to inject the `LimiterInterface` into `Limit`. We don't have to store the limiter as state but can construct the data object from it:

```php
public function __construct(int $availableTokens, \DateTimeImmutable $retryAfter, bool $accepted, LimiterInterface $limiter)
{
    $this->availableTokens = $availableTokens;
    $this->retryAfter = $retryAfter;
    $this->accepted = $accepted;
    $this->strategy = get_class($limiter);
    $this->metadata = $limiter->getMetadata();
}
```

Reply:
Don't mean to dwell on this - I really am fine with not doing this :)

Reply:
That's why I first through that limit could be created on Limiter object constructor and then be initialized with strategy and metadata. Then on each consume, you could modify other informations. @wouterj  ?

Reply:
wait i'm doing trash right now

Reply:
I honestly think it's good like this :)